src/test/regress/sql/node_conninfo_reload.sql

```diff
@@ -0,0 +1,154 @@
+-- Make sure changes citus.node_conninfo shutdown connections with old settings
+CREATE SCHEMA node_conninfo_reload;
```

Comment:
This test suite is pretty comprehensive, thanks!

Maybe one test for completeness could be:

```SQL
BEGIN;
-- successful query
-- reload that'd require connections to go away
-- a query that'd fail by itself, like SELECT count(*)/0 FROM table;
ROLLBACK
```